qiskit/tools/__init__.py

```diff
@@ -38,7 +29,34 @@
 .. automodule:: qiskit.tools.events
 
 """
-
-from .parallel import parallel_map
-from .monitor import job_monitor, backend_monitor, backend_overview
-from .events import progressbar
+import importlib
+import warnings
+
+
+_DEPRECATED_NAMES = {
+    "parallel_map": "qiskit.utils",
+    "parallel": "qiskit.utils.parallel",
+}
+
+
+_DEPRECATED_REMOVALS = {"job_monitor", "backend_monitor", "backend_overview", "progressbar"}
+
+
+def __getattr__(name):
+    if name in _DEPRECATED_NAMES:
+        module_name = _DEPRECATED_NAMES[name]
+        warnings.warn(
+            f"Accessing '{name}' from '{__name__}' is deprecated since Qiskit 0.46.0 "
+            f"and will be removed in Qiskit 1.0.0. Import from '{module_name}' instead ",
+            DeprecationWarning,
+            2,
+        )
+        return getattr(importlib.import_module(module_name), name)
```

Comment:
Even with the 0.46 cheat / clever trick, this is technically breaking our deprecation policy, because there's no way to support both 0.45 and 0.46 without getting a warning - this PR is the one that moves them.  Other things in 0.46 are deprecations for removals that are effected in the 1.0 release, where the alternative is "don't use this" rather than "use this from somewhere else".

Depending on how strict we want to be about that, we _might_ need to live with `qiskit.tools`, but I think for the 1.0 release it's probably fine.  Possibly still an edge case of the deprecation policy we ought to clarify.

Alternatively, we could remove `parallel_map` from the public API, but I imagine there are users of it, so it's suboptimal.

Reply:
Yeah, I was weighing the awkwardness of keeping the `qiskit.tools` package around with a single module vs violating the deprecation policy a bit in this case. I figured doing this all in 0.46 and violating the rules here was better than keeping `qiskit.tools` around in 1.0.0. But I'm not tied to that if people object I'm fine keeping this as `qiskit.tools` for the time being.

Reply:
I'm ok with doing it in this release - I think the set of users who might be using `parallel_map` will have a large overlap with the set of people who'll be broken for other reasons by 1.0.